QPM/Providers/RemoteQPMDependencyResolver.cs

```diff
@@ -455,6 +487,29 @@ public void ResolveDependency(in Config myConfig, in RestoredDependencyPair pair
             }
         }
 
+
+
+        private static void PlaceDep(string tempLoc, string fileLoc)
+        {
+            if (linker == null)
+            {
+                File.Copy(tempLoc, fileLoc);
+                return;
+            }
+
+            try
+            {
+                // Attempt to make symlinks to avoid unnecessary copy
+                Console.WriteLine($"Creating symlink from {tempLoc} to {fileLoc}");
+                linker.CreateLink(tempLoc, fileLoc);
```

Comment:
As far as this goes, this is _fine_, but throwing twice in general seems like a bit of a waste (also added constructors just to throw is certainly subpar). Instead, I'd recommend doing something similar to what I suggested in the last review, which was to determine whether a link will succeed or fail and then go through with it.

In this case, you can, for example:
- Change `linker` to be non-nullable, constructed in declaration
- Remove explicit static constructor from this type
- Remove throws on construction in implementing types
- Add a linker method for checking validity, which evaluates to: `_linker is not null`
- If this evaluates to false, you know the platform doesn't support it
- Otherwise, you can do it.
- Also would allow you to change this current method to avoid a try catch and instead have a `IsValid` check followed by a `CreateLink` or a fallback `File.Copy`

You can implement it in a different way, should you choose to.